osquery/database/database.cpp

```diff
@@ -343,4 +350,60 @@ void dumpDatabase() {
     }
   }
 }
+
+Status updateDatabase() {
+  std::string db_results_version{""};
+  getDatabaseValue(kPersistentSettings, "results_version", db_results_version);
+
+  if (db_results_version == kDatabseResultsVersion) {
+    return Status();
+  }
+
+  std::vector<std::string> keys;
+  auto s = scanDatabaseKeys(kQueries, keys);
+  if (!s.ok()) {
+    return Status(1, "Failed to lookup legacy query data from database");
+  }
+
+  for (const auto& key : keys) {
+    std::string value;
+    if (!getDatabaseValue(kQueries, key, value)) {
+      LOG(WARNING) << "Failed to get value from database " << key;
+      continue;
+    }
+
+    pt::ptree tree;
+    try {
+      std::stringstream ss;
+      ss << value;
+      pt::read_json(ss, tree);
+    } catch (const pt::json_parser::json_parser_error& /* e */) {
+      LOG(INFO) << "Conversion from ptree to RapidJSON failed for " << key;
+      continue;
+    }
+
+    auto json = JSON::newArray();
+    for (const auto& t : tree) {
```

Comment:
This is an issue -- we end up stomping all over other values that are expected to be ints:
Before:
```
queries[chrome_extensionscounter]: 0
queries[chrome_extensionsepoch]: 0
queries[crontab]: []
queries[crontabcounter]: 0
queries[crontabepoch]: 0
```
After:
```
...
queries[chrome_extensionscounter]: []
queries[chrome_extensionsepoch]: []
queries[crontab]: []
queries[crontabcounter]: []
queries[crontabepoch]: []
...
```

Reply:
Yes, you will need to exclude matches with the counter and epoch suffixes

Reply:
I added a check up above on the `key`:
```
if (boost::algorithm::ends_with(key, kDbEpochSuffix) ||
    boost::algorithm::ends_with(key, kDbCounterSuffix)) {
  continue;
}
```